### DISCUSSION TITLE
Team Review 2021-11-09 - AndExpand Layout options

### DISCUSSION THREAD

**Discussion by jsuarezruiz** [dotnet/maui#3324] [API Review]
The AndExpand Layout option cause a lot of confusion mostly because only works with StackLayout. We have different options:

### 1 Option - Remove AndExpand options 

Pros
- Remove the issue and reduce the confusion
- Easy to apply, just remove some values in the LayoutOptions enum

Cons
- Compiler errors, the migration tool must do some changes.
- Will not allow some scenarios mostly related with StackLayout

### 2 Option - Remove AndExpand options and create new StackLayout attached properties

Pros
- Is the most correct API option and easy to use.
- Also remove the issue (confusion).

Cons
- The migration tool require more specific logic to change not only property values also, change in the case of the StackLayout a property name and value.
- Also, this change will force some breaking changes and compiling issues in Styles. To avoid it, the migration tool must fix it too.

Example:

```
<StackLayout Expand="Horizontally|Vertically" />
<StackLayout ExpandHorizontally="True" ExpandVertically="True" />
```

### 3 Option - Use two structs

Pros
- Remove the AndExpand option from the Layouts except from StackLayout. Fix the issue and avoid the confusion.
- Is easy to implement but easier for the migration tool.
 
Cons
- Have two different structs but mostly for the same and with similar content.
- Could cause compiling errors creating Styles etc? (different types).
- We need to improve intellisense with this specific scenario, to avoid the previous point.

### 4 Option - Not changes

Pros
- Not require effort.

Cons
- We are not solving the confusion issues.

**Comment by wuyuansushen** [comment]
I have learning the `StackLayout` now.
The Expansion have no effect while I following the Xamarin Tutorial [here](https://docs.microsoft.com/en-us/xamarin/get-started/tutorials/stacklayout/?tabs=vswin&tutorial-step=3) in MAUI.
Have you team decided which options will be applied now?

**Reply by Dreamescaper** [reply]
See this PR: https://github.com/dotnet/maui/pull/3362

**Reply by wuyuansushen** [reply]
Thanks.
